### PR TITLE
fix: update C implementation for `stats/base/dists/kumaraswamy/median`

### LINKED ISSUE #5047: [BUG]: C implementation for `stats/base/dists/kumaraswamy/median` has different results than the JavaScript implementation
### Description

Pointed out by @Neerajpathak07 and @yuvi-mittal

For context, see [comment](https://github.com/stdlib-js/stdlib/pull/5019#issuecomment-2631770211)

When running native tests for `stats/base/dists/kumaraswamy/median`, the majority of tests fail because the C implementation is generalized for a single value of `a = 1.0` rather than any positive number.

The fix is to change 

https://github.com/stdlib-js/stdlib/blob/367ec2c5fe4a11f2e9a450ab4b21ea66db002cc2/lib/node_modules/%40stdlib/stats/base/dists/kumaraswamy/median/src/main.c#L42

to:

```c
return stdlib_base_pow( 1.0 - stdlib_base_pow( 2.0, -1.0 / b ), 1.0 / a );
```

### Related Issues

None

### Questions

No.

### Demo

_No response_

### Reproduction

```shell
- Run `make install-node-addons NODE_ADDONS_PATTERN="stats/base/dists/kumaraswamy/median"`
- Run `make test TESTS_FILTER=".*/stats/base/dists/kumaraswamy/median/test/test.native.js"`
```

### Expected Results

```shell
total:     1009
passing:   1009
```

### Actual Results

```shell
total:     1009
passing:   27
failing:   982
```

### Version

_No response_

### Environments

N/A

### Browser Version

_No response_

### Node.js / npm Version

_No response_

### Platform

_No response_

### Checklist

- [x] Read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.

### PR DESCRIPTION
**Pull request by Krishna-Sharma-g**
Resolves #5047 .

## Description
#5047 

> What is the purpose of this pull request?

This pull request:

fixes the bug of   failing testcases in the kumarswamy/median/src/main.c   and also improved and matched the test.native.js and test.js  

- 1. Fixed the incorrectly named test that was referring to "gamma" instead of "a" by completely replacing it with the proper test for nonpositive "a" parameter from test.js.

- 2.Added the missing test for nonpositive "b" parameter, copying the test cases from test.js.

- 3.Ensured all edge cases are properly tested with the same combinations of parameter values as in test.js.
Maintained the native-specific parts like the tryRequire setup and opts variable for test skipping.

## Related Issues

> Does this pull request have any related issues?

no

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

image

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

- [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Krishna-Sharma-g** [comment]
hello @anandkaranubc please review once

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/src/main.c [hunk 1]
```diff
@@ -39,5 +39,5 @@ double stdlib_base_dists_kumaraswamy_median( const double a, const double b ) {
 	) {
 		return 0.0/0.0; // NaN
 	}
-	return 1.0 - stdlib_base_pow( 0.5, 1.0 / b );
+	return stdlib_base_pow( 1.0 - stdlib_base_pow( 2.0, -1.0 / b ), 1.0 / a );
```
Comment (anandkaranubc) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/src/main.c]:
```suggestion
	return stdlib_base_pow( 1.0 - stdlib_base_pow( 2.0, -1.0/b ), 1.0/a );
```

**Comment by anandkaranubc** [comment]
@Krishna-Sharma-g, thanks for opening this PR. As I mentioned earlier in the issue, there are many more changes required for the current package `stats/base/dists/kumaraswamy/median`.

I have already created a commit [here](https://github.com/stdlib-js/stdlib/commit/90dab8dc447c898f6fe629f32bb924efd3eb6e79) that will guide you on what other changes need to be made. You can just copy all the changes from there and update this PR. Cheers!

**Review by anandkaranubc — CHANGES_REQUESTED** [review]

**Comment by anandkaranubc** [comment]
Can you also change the title to:

fix: update C implementation for `stats/base/dists/kumaraswamy/median`

Note that there are backticks around stats/base/dists/kumaraswamy/median

**Comment by anandkaranubc** [comment]
I also noticed that your current PR is based off your develop branch. While you can keep it like this for the current PR, I would suggest creating a new branch each time for future changes/PRs.

**Comment by Krishna-Sharma-g** [comment]
> @Krishna-Sharma-g, thanks for opening this PR. As I mentioned earlier in the issue, there are many more changes required for the current package `stats/base/dists/kumaraswamy/median`.
> 
> I have already created a commit [here](https://github.com/stdlib-js/stdlib/commit/90dab8dc447c898f6fe629f32bb924efd3eb6e79) that will guide you on what other changes need to be made. You can just copy all the changes from there and update this PR. Cheers!

okay @anandkaranubc  thankyou for that

**Comment by Krishna-Sharma-g** [comment]
> Can you also change the title to:
> 
> fix: update C implementation for `stats/base/dists/kumaraswamy/median`
> 
> Note that there are backticks around stats/base/dists/kumaraswamy/median

done !

**Comment by Krishna-Sharma-g** [comment]
> I also noticed that your current PR is based off your develop branch. While you can keep it like this for the current PR, I would suggest creating a new branch each time for future changes/PRs.

okay next time i will be aware of that

**Comment by Krishna-Sharma-g** [comment]
i

**Comment by Krishna-Sharma-g** [comment]
@anandkaranubc does this pr need any more changes or it is ready to merge

**Comment by anandkaranubc** [comment]
> @anandkaranubc does this pr need any more changes or it is ready to merge

@Krishna-Sharma-g Everything, that was made in this commit [here](https://github.com/stdlib-js/stdlib/commit/90dab8dc447c898f6fe629f32bb924efd3eb6e79)

**Comment by Krishna-Sharma-g** [comment]
@anandkaranubc oo sorry i missed that

**Comment by Krishna-Sharma-g** [comment]
hello @anandkaranubc i synced the file and due to that it commited the 
Merge branch 'stdlib-js:develop' into develop   so do i have to do something or it is okay

**Comment by anandkaranubc** [comment]
There aren't any "file" changes in the PR, so there's nothing to worry about.

**Comment by Krishna-Sharma-g** [comment]
@anandkaranubc  changes done !

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c [hunk 2]
```diff
@@ -93,9 +94,10 @@ static double random_uniform( const double min, const double max ) {
 * @return elapsed time in seconds
 */
 static double benchmark( void ) {
-	double elapsed;
+
```
Comment (anandkaranubc) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c]: No need of an empty line here. Please remove it.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c [hunk 3]
```diff
@@ -1,3 +1,4 @@
...
```
Comment (anandkaranubc) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c]: Same comment

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/lib/native.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (anandkaranubc) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/lib/native.js]: Same comment

**Comment by Krishna-Sharma-g** [comment]
@anandkaranubc on it

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json [hunk 5]
```diff
@@ -40,8 +40,8 @@
       "dependencies": [
         "@stdlib/math/base/napi/binary",
         "@stdlib/math/base/assert/is-nan",
-        "@stdlib/math/base/special/pow",
-        "@stdlib/constants/float64/eps"
+        "@stdlib/math/base/special/pow"
+
```
Comment (anandkaranubc) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json]: Same comment

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json [hunk 6]
```diff
@@ -74,8 +74,8 @@
       "libpath": [],
       "dependencies": [
         "@stdlib/math/base/assert/is-nan",
-        "@stdlib/math/base/special/pow",
-        "@stdlib/constants/float64/eps"
+        "@stdlib/math/base/special/pow"
+
```
Comment (anandkaranubc) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json]: Same comment

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c [hunk 7]
```diff
@@ -93,9 +93,9 @@ static double random_uniform( const double min, const double max ) {
 * @return elapsed time in seconds
 */
 static double benchmark( void ) {
-	double elapsed;
-	double a[ 100 ];
+    double a[ 100 ];
```
Comment (anandkaranubc) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c]: Can you fix the alignment here? It should be tab spaced

Comment (Krishna-Sharma-g) [reply lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c]: done

**Comment by anandkaranubc** [comment]
@kgryte, I think this is ready for merge. As you mentioned, there are many more changes than initially stated in the issue. Hopefully, we've covered them all. Thanks!

**Review by Planeshifter — DISMISSED** [review]
Thanks @Krishna-Sharma-g for your contribution and @anandkaranubc for initial review! Will merge PR shortly.

**Review by Planeshifter — APPROVED** [review]
